torch/_prims/__init__.py

```diff
@@ -1815,14 +1815,15 @@ def _as_strided_scatter_meta(
     utils.validate_shape(size)
     utils.validate_strides(stride)
 
-    required_size = utils.compute_required_storage_length(size, stride, storage_offset)
+    input_length = utils.compute_required_storage_length(input.size(), input.stride(), input.storage_offset())
+    output_length = utils.compute_required_storage_length(size, stride, storage_offset)
     utils.check(
-        input.numel() >= required_size,
+        input_length >= output_length,
```

Comment:
Do you have a repro using functionalization?

Reply:
updated with repro